xdsl/dialects/pdl.py

```diff
@@ -182,6 +183,9 @@ class ApplyNativeConstraintOp(IRDLOperation):
 
     name = "pdl.apply_native_constraint"
     constraint_name = prop_def(StringAttr, prop_name="name")
+    is_negated = prop_def(
+        BoolAttr, prop_name="isNegated", default_value=BoolAttr.from_bool(False)
+    )
```

Comment:
Can you add a test, please? Also with a roundtrip with MLIR?

Reply:
(I have to add a test with MLIR still)

Reply:
In MLIR, `pdl.rewrite`'s region isn't optional:
```
let regions = (region AnyRegion:$bodyRegion);

let assemblyFormat = [{
    ($root^)? (`with` $name^ (`(` $externalArgs^ `:` type($externalArgs) `)`)?)?
              ($bodyRegion^)?
    attr-dict-with-keyword
  }];
```
But the assembly format for the region is an optional group meaning it shouldn't get printed when the region is empty (default case).
https://mlir.llvm.org/docs/DefiningDialects/Operations/#optional-groups
> When a non-variadic region anchors a group, the detector for printing the group is if the region is empty. 

This behaviour seems to be missing in xDSL?

(This is causing difficulties with an interop test)

Reply:
Oh it seems to already be working for parsing, but not for printing, if I'm reading the code correctly. What happens when you use this format?

Reply:
OK no I see it's a little weirder than this... I'm looking into it

Reply:
Merged!